src/main/java/edu/gatech/cs2340/trydent/animation/KeyframeAnimationBuilder.java

```diff
@@ -0,0 +1,303 @@
+        buildingFrame.setPosition(orientation.getPosition());
+        buildingFrame.setRotation(orientation.getRotation());
+        buildingFrame.setScale(orientation.getScale());
+        return this;
+    }
+
+    /**
+     * Sets the interpolation strategy used to generate in-between frames. This
+     * defaults to Interpolation.SMOOTH and for most uses will not need to be
+     * changed.
+     *
+     * @param interpolation
+     * @return
+     */
+    public KeyframeAnimationBuilder setInterpolation(Interpolation<BaseVector<?>> interpolation) {
+        buildingFrame.interpolation = interpolation;
+        return this;
+    }
+
+    /**
+     * Adds a new keyframe at the current position, rotation, and scale.
+     *
+     * @param duration
+     *            - Time in seconds it should take for the animation to get from
+     *            this keyframe to the one after it. For the final keyframe,
+     *            this value is ignored, as the final keyframe has no duration.
+     * @return
+     */
+    public KeyframeAnimationBuilder addKeyframe(double duration) {
+        buildingFrame.duration = duration;
+        keyframes.add(buildingFrame);
+        Keyframe next = new Keyframe();
+        next.interpolation = buildingFrame.interpolation;
+        next.setPosition(buildingFrame.getPosition());
+        next.setRotation(buildingFrame.getRotation());
+        next.setScale(buildingFrame.getScale());
+        buildingFrame = next;
+        return this;
+    }
+
+    /**
+     * Adds a new keyframe of zero duration at the current position, rotation,
+     * and scale.
+     * <p>
+     * Unless this is the final keyframe in the animation, this call should be
+     * followed by a call to setTotalDuration(duration), which will
+     * automatically change the duration of this frame and other zero-duration
+     * frames to appropriately fill in the remaining time.
+     *
+     * @return
+     */
+    public KeyframeAnimationBuilder addKeyframe() {
+        return addKeyframe(0);
+    }
+
+    /**
+     * Sets the duration for the /TOTAL/ animation, changing the duration of all
+     * individual keyframes that have been set so far to accomplish this.
+     *
+     * If any keyframes have a negative or 0 duration, they will be used to fill
+     * in the gaps in time, and the other frames will be left alone.
+     *
+     * @param duration
+     * @return
+     */
+    public KeyframeAnimationBuilder setTotalDuration(double duration) {
+        if (duration <= 0) {
+            throw new TrydentException("The duration of the animation must be a positive number!");
+        }
+        enforcedTotal = duration;
+        return this;
+    }
+
+    /**
+     * If true, this animation will be constructed so that it loops back to the
+     * beginning when it ends.
+     *
+     * @param circular
+     * @return
+     */
+    public KeyframeAnimationBuilder setCircular(boolean circular) {
```

Comment:
It's not really a field that can be changed once the animation is built, so it either would have to be here or as a parameter to build(). In the latter case, I'd have to have lots of overridden versions of build() for different optional parameters, and the whole point of method chaining is to avoid that. However, it might be a good idea to rename the method, or at least change the docs, to make it clear that it affects the entire animation, not just the single keyframe.